### PR TITLE
analyze: allow a custom policy to skip a check

### PR DESCRIPTION
**Pull request by bluca**
also improve the manpage for existing options

### CODE REVIEW

File: man/systemd-analyze.xml [hunk 1]
```diff
@@ -1075,9 +1075,13 @@ Service b@0.service not loaded, b.socket cannot be started.
...
           as default. The weight and range fields are used in determining the overall exposure level
-          of the unit files so by allowing users to manipulate these fields, 'security' gives them
-          the option to decide for themself which ids are more important and hence, should have a greater
-          effect on the exposure level. </para>
+          of the unit files: the value of each setting is assigned a badness score, which is multiplied
+          by the policy weight and divided by the policy range to determine the overall exposure that
+          the setting implies. The computed badness is summed across all settings in the unit file,
+          normalized to the 1…100 range, and used to determine the overall exposure level of the unit.
+          By allowing users to manipulate these fields, the 'security' verb gives them the option to
+          decide for themself which ids are more important and hence, should have a greater effect on
```
Comment (poettering) [inline man/systemd-analyze.xml]: the comma is weird, no?

File: src/analyze/analyze-security.c [hunk 2]
```diff
@@ -1764,6 +1779,11 @@ static int assess(const SecurityInfo *info,
                         d = strdup("Service runs in special boot phase, option is not appropriate");
                         if (!d)
                                 return log_oom();
+                } else if (access_skip(a, policy)) {
+                        badness = UINT64_MAX;
+                        d = strdup("Option excluded by policy, skipping");
+                        if (!d)
+                                return log_oom();
```
Comment (poettering) [inline src/analyze/analyze-security.c:1771]: why a separate field for this? wouldn't a weight of zero more naturally and less redundantly encode this?

**Comment by poettering** [comment]
looks ok to me, but I think it would be nicer to use an existing field for this, and make 0 (or `null`) trigger this behaviour. Or what am I missing?

**Comment by bluca** [comment]
> looks ok to me, but I think it would be nicer to use an existing field for this, and make 0 (or `null`) trigger this behaviour. Or what am I missing?

Yeah no particular reason, that works too. Updated.

File: man/systemd-analyze.xml [hunk 3]
```diff
@@ -1081,7 +1081,8 @@ Service b@0.service not loaded, b.socket cannot be started.
           normalized to the 1…100 range, and used to determine the overall exposure level of the unit.
           By allowing users to manipulate these fields, the 'security' verb gives them the option to
           decide for themself which ids are more important and hence should have a greater effect on
-          the exposure level.</para>
+          the exposure level. A weigth of <literal>0</literal> means the setting will not be
```
Comment (poettering) [inline man/systemd-analyze.xml]: weight

Comment (bluca) [reply man/systemd-analyze.xml]: ops, fixed

**Pull request merged**